compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4916,11 +4919,7 @@ object Types {
      *  is also a singleton type.
      */
     def instantiate(fromBelow: Boolean)(using Context): Type =
-      val tp = TypeComparer.instanceType(origin, fromBelow, widenUnions, nestingLevel)
-      if myInst.exists then // The line above might have triggered instantiation of the current type variable
```

Comment:
This check used to be necessary so I would prefer to keep it now that the `assert` in `instantiateWith` has been added back, unless we have some strong reason to believe it no longer is.